format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingConfig.java

```diff
@@ -114,19 +115,21 @@ public Schema getSchema() {
    * @param job job to retrieve the file system
    * @return {@link Path}
    */
-  public Path getFilePathForSchemaGeneration(String path, String regexPathFilter, Configuration configuration, Job job)
+  public ArrayList<Path>
```

Comment:
Does it need to be an ArrayList? If not can we change the signature to return a List instead

Reply:
+1. We should always return the Interface